backend/src/data-analyzer/criteria-based-analyzers/__tests__/contains-condition.spec.ts

```diff
@@ -0,0 +1,111 @@
+import { AstNodeType, GeneralAst } from "src/ast/types/general-ast";
+import { ExpressionNodeType } from "src/ast/types/general-ast/ast-nodes/expression-node";
+import { StatementNodeType } from "src/ast/types/general-ast/ast-nodes";
+import { containsCondition } from "../contains-condition";
+
+describe("Criteria Based Analyzer", () => {
+  describe("containsCondition", () => {
+    it("returns true if the AST contains a condition", () => {
+      const output = containsCondition(
```

Comment:
Shouldn't we be able to specify the condition? 🤔 
I guess that's for a subsequent PR 😁